### PR TITLE
cleanup references to old build code

### PR DESCRIPTION
**Pull request by jjergus**
README + userdata code branches that were only reachable from the old scripts

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,51 +75,32 @@ Building source tarball and linux packages for new releases
 -----------------------------------------------------------
 
  - tag the release on the `hhvm-staging` repo
- - `bin/make-all-packages-on-aws VERSION`
+ - `bin/build-on-aws VERSION`
 
 If you just need to rebuild for one distribution, with no code changes:
 
- - edit `DISTRO/PKGVER` if needed (e.g. if a previouis package was pushed for this HHVM version and distribution) and push to github
- - `bin/make-package-on-aws VERSION DISTRO`
```
Comment (fredemmott) [inline README.md]: is the azure script still meant to be used directly for mac?

File: aws/bin/publish-release-source [hunk 2]
```diff
@@ -1,4 +1,4 @@
-!/bin/bash
```
Comment (fredemmott) [inline aws/bin/publish-release-source]: heh, this is before the `set -e` :D

File: aws/bin/update-repos [hunk 3]
```diff
@@ -17,24 +17,6 @@ if ! gpg --list-secret-key opensource+hhvm@fb.com >/dev/null 2>/dev/null; then
...
-  aws ec2 attach-volume \
-    --device /dev/sdf \
-    --instance-id "$INSTANCE_ID" \
-    --volume-id "$VOLUME_ID"
-  sleep 10 # wait for EC2 to attach
-
-  mkdir /mnt/dl.hhvm.com
-  mount /dev/xvdf /mnt/dl.hhvm.com
-  MOUNTED="1"
-fi
```
Comment (fredemmott) [inline aws/bin/update-repos]: this change looks unrelated?

Comment (jjergus) [reply aws/bin/update-repos]: Same as the other changes -- this conditional branch of the script was only reached from the old build system. The new one mounts the drive from the [init script](https://github.com/hhvm/packaging/blob/master/aws/hhvm1/worker/init/update-repos.sh) (no need to unmount and re-mount it between each version being published, especially since that is the part that used to randomly fail from time to time).

File: README.md (hunk 1 above)
Comment (jjergus) [reply README.md]: I guess the azure script is not needed anymore, since the builds can be triggered through AWS (`bin/build-on-aws BuildAndPublishMacOS $VERSION` works), but running the AWS state machine just to build a Mac binary has a lot of overhead so it's probably still useful to keep the azure script too.
